llvm/unittests/CAS/ThreadSafeAllocatorTest.cpp

```diff
@@ -0,0 +1,33 @@
+//===- llvm/unittest/CAS/ThreadSafeAllocatorTest.cpp ----------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/CAS/ThreadSafeAllocator.h"
+#include "llvm/Support/ThreadPool.h"
+#include "gtest/gtest.h"
+#include <thread> 
+
+using namespace llvm;
+
+TEST(ThreadSafeAllocatorTest, Basics) {
+  cas::ThreadSafeAllocator<BumpPtrAllocator> Alloc;
+  ThreadPool Threads;
+
+  for (unsigned Index = 1; Index < 100; ++Index)
+    Threads.async(
+        [&Alloc](unsigned I) {
+          int *P = (int *)Alloc.Allocate(sizeof(int) * I, alignof(int));
+          P[I - 1] = I;
+        },
+        Index);
+
```

Comment:
Is it worth having a second test that covers the single-parameter `Allocate()`? Tests could be called "Allocate" and "AllocateWithAlign".